src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/TextAreaFilter.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Filter\AttributeFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+use Pim\Component\Catalog\Validator\AttributeValidatorHelper;
+
+/**
+ * Text area Filter for an Elasticsearch query
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class TextAreaFilter extends AbstractFilter implements AttributeFilterInterface
+{
+    /**
+     * @param AttributeValidatorHelper $attrValidatorHelper
+     * @param array                    $supportedAttributeTypes
+     * @param array                    $supportedOperators
+     */
+    public function __construct(
+        AttributeValidatorHelper $attrValidatorHelper,
+        array $supportedAttributeTypes = [],
+        array $supportedOperators = []
+    ) {
+        $this->attrValidatorHelper = $attrValidatorHelper;
+        $this->supportedAttributeTypes = $supportedAttributeTypes;
+        $this->supportedOperators = $supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeFilter(
+        AttributeInterface $attribute,
+        $operator,
+        $value,
+        $locale = null,
+        $scope = null,
+        $options = []
+    ) {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkLocaleAndScope($attribute, $locale, $scope);
+
+        if (Operators::IS_EMPTY !== $operator && Operators::IS_NOT_EMPTY !== $operator) {
+            $this->checkValue($attribute, $value);
+            $value = $this->escapeValue($value);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $scope);
+
+        switch ($operator) {
+            case Operators::STARTS_WITH:
+                $attributePath .= '.raw';
+                $clause = [
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => $value . '*',
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::CONTAINS:
+                $clause = [
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => '*' . $value . '*',
```

Comment:
I took what ES bundle tests indicated. https://github.com/akeneo/ElasticSearchBundle/blob/master/Resources/doc/queries_test/features/querying_text_area_field.feature#L53
And also what the StringFilter in ES bundle was doing: https://github.com/akeneo/ElasticSearchBundle/blob/master/Query/Filter/StringFilter.php#L99